lesson20.md

```diff
@@ -0,0 +1,319 @@
+<!-- .slide: id="lesson20" -->
+
+# Basic Frontend - Spring 2021
+
+Lesson 20, Tuesday, 2021-05-18
+
+---
+
+### Recap - Data types
+
+```js
+let a1 = 42;
+let a2 = true;
+let a3 = "hello";
+let a4 = function() {};
+let a5;
+let a6 = [];
+let a7 = {};
+```
+
+---
+
+### Solution - Data types
+
+```js
+let a1 = 42;       // number
+let a2 = true;     // boolean
+let a3 = "hello";  // string
+let a4 = function() {};  // function
+let a5;            // undefined
+let a6 = [];       // object (array)
+let a7 = {};       // object
+```
+
+---
+
+### Null
+
+* There's one datatype that we haven't learned about yet: `null`
+* `null` is like `undefined`. It indicates the absence of a value.
+* `null` can be used to indicate an *intentional* absence of an object value.
+* Example: `document.getElementById("NoExist");` returns `null`
+
+---
+
+### Recap - DOM
+
+* DOM is an API
+* It allows us to add/remove/change any element of a web page
+* We can make TODO lists and cool web shops
+* But how would we create a web page with "dynamic" data, for example today's weather?
+
+---
+
+<!-- .slide: id="WebAPI" -->
+
+# Web APIs
+
+---
+
+### Basic building blocks
+
+We need to first learn about a few concepts:
```

Comment:
Maybe mention that we can exchange one method with another one while keeping the others the same. For example JSON format can be exchanged with XML while fetch and REST stays the same. Or fetch can be exchanged with XMLHttpRequest. So this separation of data format from protocol from server connection allows flexibility to a programmer. But this combination is the most common one you can see in Webapps.